### PR TITLE
Change default comment token to # for unrecognized files

### PR DESCRIPTION
**Pull request by kfatyuip**
Helix currently uses // as the default comment token for unrecognized file types, but this doesn’t align well with Linux conventions. Since Linux configuration files—such as locale.conf, pacman.conf, and /etc/sudoers—almost always use # for comments, defaulting to // feels out of place.

Files like locale.conf and pacman.conf are sometimes misclassified as HOCON simply because they have a .conf extension, and files without extensions (e.g., sudoers) end up with // as their comment token, which doesn’t match their typical usage.

Most programming languages that use // (e.g., Rust, JavaScript) are easily recognized by their file extensions, whereas files without extensions are more likely to follow Linux conventions and use #. Switching the default to # makes Helix more predictable and practical for Linux users.

### CODE REVIEW

File: helix-core/src/comment.rs [hunk 1]
```diff
@@ -345,17 +345,17 @@ mod test {
 
             let text = doc.slice(..);
 
-            let res = find_line_comment("//", text, 0..3);
+            let res = find_line_comment("#", text, 0..3);
```
Comment (TornaxO7) [inline helix-core/src/comment.rs:348]: Hm... I don't really see the need of these kind of changes. It should be enough to just change the constant as you did.
May I ask why you did that?

Comment (kfatyuip) [reply helix-core/src/comment.rs:348]: hmm, i ignored to change the token in my first commit and failed to pass [the workflow test suite](https://github.com/helix-editor/helix/actions/runs/11877333431/job/33096665578), so i replaced all the token from "//" to "#" in the tests

Comment (kfatyuip) [reply helix-core/src/comment.rs:348]: > Hm... I don't really see the need of these kind of changes. It should be enough to just change the constant as you did. May I ask why you did that?

hmm, i ignored to change the token in my first commit and failed to pass [the workflow test suite](https://github.com/helix-editor/helix/actions/runs/11877333431/job/33096665578), so i replaced all the token from "//" to "#" in the tests

Comment (TornaxO7) [reply helix-core/src/comment.rs:348]: That's really weird. If I run `cargo integration-test` on my machine with the following test function:
```rust
        #[test]
        fn not_commented() {
            // four lines, two space indented, except for line 1 which is blank.
            let doc = Rope::from("  1\n\n  2\n  3");

            let text = doc.slice(..);

            let res = find_line_comment("//", text, 0..3);
            // (commented = false, to_change = [line 0, line 2], min = col 2, margin = 0)
            assert_eq!(res, (false, vec![0, 2], 2, 0));
        }
```
everything works fine, as expected.

Comment (kfatyuip) [reply helix-core/src/comment.rs]: okay there's the new comment 02162e998d9cf7815963786f790c0e0c22f3ea24, just keep it as it was

Comment (TornaxO7) [reply helix-core/src/comment.rs]: > that's not weird

well, not for all tests but for this one, it is. There are no comment tokens, so the result should be the same that there isn't a comment token.

> other tests failed due to "//" is not replaced

aye

Comment (TornaxO7) [reply helix-core/src/comment.rs]: > there's no reason to keep "//" here

hm... maybe for testing the `find_line_comment` for different comment tokens as well?

Anyways, you already applied the changes :)

**Review by TornaxO7 — APPROVED** [review]

**Review by the-mikedavis — APPROVED** [review]
The reasoning makes sense to me. Previously I was thinking we might want to drop the "default comment token" behavior and refuse to toggle anything for text languages on `C-c` but toggling `#` is probably a more reasonable default behavior.

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**